app/src/main/java/tech/relaycorp/cogrpc/server/CogRPCServer.kt

```diff
@@ -45,14 +43,19 @@ internal constructor(
         withContext(Dispatchers.IO) {
             setupTLSProvider()
 
+            val certGenerator = TLSCertificateGenerator.generate()
```

Comment:
Can we inject this somehow through the constructor, or at least make it internally swappable? It would make future testing easier.

Reply:
What do you mean by internally swappable?

I did consider moving `TLSCertificateGenerator.generate()` to a lazily-initialised field because I thought that in the future we might not want to use TLS (during local dev), but I forgot to follow through.

Reply:
Internally swappable would be to make the generator an `internal var` that can be swapped. For example:
```
internal var tlsCertificateGeneratorProvider: (() -> CertGenerator) = TLSCertificateGenerator::generate
```

By default it would use what we setup, but in a test we could switch it to another CertGenerator.

Reply:
Gotcha. Makes sense :+1: I'll change it.